x-pack/plugin/data-science/src/test/java/org/elasticsearch/datascience/cumulativecardinality/CumulativeCardinalityAggregatorTests.java

```diff
@@ -0,0 +1,249 @@
+        "2017-01-03T05:11:31", //1
+        "2017-01-05T08:24:05", //5
+        "2017-01-05T13:09:32", //1
+        "2017-01-07T13:47:43", //7
+        "2017-01-08T16:14:34", //1
+        "2017-01-09T17:09:50", //9
+        "2017-01-09T22:55:46");//10
+
+    private static final List<Integer> datasetValues = Arrays.asList(1,1,3,1,5,1,7,1,9,10);
+    private static final List<Double> cumulativeCardinality = Arrays.asList(1.0,1.0,2.0,2.0,3.0,3.0,4.0,4.0,6.0);
+
+    // Initialize plugin so we can set license state
+    static DataSciencePlugin PLUGIN = new DataSciencePlugin() {
+        @Override
+        protected void registerLicenseListener() {
+            DataSciencePlugin.isDataScienceAllowed = true;
+        }
+    };
+
+    public void testSimple() throws IOException {
+
+        Query query = new MatchAllDocsQuery();
+
+        DateHistogramAggregationBuilder aggBuilder = new DateHistogramAggregationBuilder("histo");
+        aggBuilder.calendarInterval(DateHistogramInterval.DAY).field(HISTO_FIELD);
+        aggBuilder.subAggregation(new CardinalityAggregationBuilder("the_cardinality", ValueType.NUMERIC).field(VALUE_FIELD));
+        aggBuilder.subAggregation(new CumulativeCardinalityPipelineAggregationBuilder("cumulative_card", "the_cardinality"));
+
+        executeTestCase(query, aggBuilder, histogram -> {
+            assertEquals(9, ((Histogram)histogram).getBuckets().size());
+            List<? extends Histogram.Bucket> buckets = ((Histogram)histogram).getBuckets();
+            int counter = 0;
+            for (Histogram.Bucket bucket : buckets) {
+                assertThat(((InternalSimpleLongValue) (bucket.getAggregations().get("cumulative_card"))).value(),
+                    equalTo(cumulativeCardinality.get(counter)));
+                counter += 1;
+            }
+        });
+    }
+
+    public void testAllNull() throws IOException {
+        Query query = new MatchAllDocsQuery();
+
+        DateHistogramAggregationBuilder aggBuilder = new DateHistogramAggregationBuilder("histo");
+        aggBuilder.calendarInterval(DateHistogramInterval.DAY).field(HISTO_FIELD);
+        aggBuilder.subAggregation(new CardinalityAggregationBuilder("the_cardinality", ValueType.NUMERIC).field("foo"));
+        aggBuilder.subAggregation(new CumulativeCardinalityPipelineAggregationBuilder("cumulative_card", "the_cardinality"));
+
+        executeTestCase(query, aggBuilder, histogram -> {
+            assertEquals(9, ((Histogram)histogram).getBuckets().size());
+            List<? extends Histogram.Bucket> buckets = ((Histogram)histogram).getBuckets();
+            for (Histogram.Bucket bucket : buckets) {
+                assertThat(((InternalSimpleLongValue) (bucket.getAggregations().get("cumulative_card"))).value(), equalTo(0.0));
+            }
+        });
+    }
+
+    /**
+     * The validation should verify the parent aggregation is allowed.
+     */
+    public void testValidate() throws IOException {
```

Comment:
This test method doesn't assert anything.  If this is just a "make sure it doesn't throw" test, maybe that could be rolled into a data test instead?

Reply:
Hmm yeah.  I struggled how to fix this, there really isn't a good place elsewhere to incorporate it because it relies on parents to validate (and the agg factory in particular, which gets built in the super AggregatorTestClass method).  Could refactor things but it's a lot of gymnastics.

I instead merged the two validation methods, test all three of the "good" parents instead of randomizing, and end with the failing validation so at least there is now an assert for the test.  

Open to suggestions!